### PR TITLE
Add support for legacy files rebased

### PR DESCRIPTION
**Pull request by jcfr**

### CODE REVIEW

**Comment by jcfr** [comment]
@nicain @ajtritt The topic adding support for reading legacy NWB file as been rebased and his ready for final review.

* The notebook successfully execute from end-to-end
* All test pass (beside of few expected warnings)
* most of (if not all) "legacy" code paths have been marked using a commend `# LEGACY`

I also added a last commit re-organizing the code of the validate CLI.

@nicain when running the CLI by hand, there is a validation error (also happening with `test_legacy_2`). Is this expected ?

```
$ cd pynwb
$ pip install -e .
$ python -m pynwb.validate ./tests/integration/ui_write/496908818s.nwb
/home/jcfr/Projects/pynwb/src/pynwb/form/build/builders.py:200: UserWarning: 'comments' already exists as attributes; skipping...
  warnings.warn("'%s' already exists as %s; skipping..." % (name, self.obj_type[name]))
Traceback (most recent call last):
  File "/usr/lib/python3.5/runpy.py", line 170, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.5/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/jcfr/Projects/pynwb/src/pynwb/validate.py", line 58, in <module>
    main()
  File "/home/jcfr/Projects/pynwb/src/pynwb/validate.py", line 52, in main
    errors = validate(io)
  File "/home/jcfr/Projects/pynwb/src/pynwb/form/utils.py", line 314, in func_call
    return func(**parsed['args'])
  File "/home/jcfr/Projects/pynwb/src/pynwb/__init__.py", line 175, in validate
    return validator.validate(builder)
  File "/home/jcfr/Projects/pynwb/src/pynwb/form/utils.py", line 306, in func_call
    return func(self, **parsed['args'])
  File "/home/jcfr/Projects/pynwb/src/pynwb/form/validate/validator.py", line 199, in validate
    raise ValueError(msg)
ValueError: builder must have data type defined with attribute 'neurodata_type'
```

**Comment by nicain** [comment]
@jcfr 

> @nicain when running the CLI by hand, there is a validation error (also happening with test_legacy_2). Is this expected ?

To my understanding, yes, this is expected.  496908818s.nwb is a 1.06 file, and so won't validate to the new schema.  @ajtritt do you agree?

**Comment by nicain** [comment]
I just tested the read ability of to https://github.com/NeurodataWithoutBorders/pynwb/commits/8c077bb76f6284dddce65953fe3fd73d22a9e91f to read all Brain Observatory files currently released; everything looks good!

File: src/pynwb/form/build/map.py [hunk 1]
```diff
@@ -215,13 +215,17 @@ def __init__(self, **kwargs):
         self.__carg2spec = dict()
         self.__map_spec(spec)
 
+    def hack_get_subspec_values(self, *args, **kwargs):
```
Comment (nicain) [inline src/pynwb/form/build/map.py:218]: Making a note that we should come back in and rework get_subspec_values.  Ill make an issue for this.

**Review by nicain — APPROVED** [review]
Everything I expected is there.  Also, I confirmed that all released Brain Observatory files can be read/write round-tripped.

**Comment by jcfr** [comment]
Thanks for the review. Since Appveyor and CircleCI are green and the branch has been manually tested, I am integrating.

**Comment by jcfr** [comment]
Thanks for the review. Since Appveyor and CircleCI are green and the branch has been manually tested, I am integrating.

**Comment by jcfr** [comment]
Integrated in https://github.com/NeurodataWithoutBorders/pynwb/commit/936c4383415d335ef2f901f040c02e843ca8c2a8

**Pull request closed without merging**